Sources/EnumKit/Extensions/Combine+CaseAccessible.swift

```diff
@@ -0,0 +1,212 @@
+//
+//  Combine+CaseAccessible.swift
+//  EnumKit
+//
+//  Created by Giuseppe Lanza on 17/08/2019.
+//
+
+#if canImport(Combine)
+import Combine
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publisher where Output: CaseAccessible {
+    
+    /// Republishes all elements that match a provided case.
+    /// - parameter case: An enum case to test each source element for a matching condition.
+    /// - returns: A publisher that republishes all elements that match the case.
+    func filter(case: Output) -> Publishers.Filter<Self> {
+        return filter { $0 ~= `case` }
+    }
+    
+    /// Republishes all elements that match a provided case.
+    /// - parameter case: An enum case to test each source element for a matching condition.
+    /// - returns: A publisher that republishes all elements that match the case.
+    func filter<AssociatedValue>(case pattern: @escaping (AssociatedValue) -> Output) -> Publishers.Filter<Self> {
+        return filter { $0 ~= pattern }
+    }
+    
+    /// Republishes all elements that do not match a provided case.
+    /// - parameter case: An enum case to test each source element for a matching condition.
+    /// - returns: A publisher that republishes all elements that do not match the case.
+    func exclude(case: Output) -> Publishers.Filter<Self> {
+        return filter { $0 !~= `case` }
+    }
+    
+    /// Republishes all elements that do not match a provided case.
+    /// - parameter case: An enum case to test each source element for a matching condition.
+    /// - returns: A publisher that republishes all elements that do not match the case.
+    func exclude<AssociatedValue>(case pattern: @escaping (AssociatedValue) -> Output) -> Publishers.Filter<Self> {
+        return filter { $0 !~= pattern }
+    }
+    
+    /// Projects each matching enum case of a Publisher into its associated value.
```

Comment:
There is no associated value here, so maybe it's worth to describe that for such enums it would passed through as Void